node/pkg/driver/node_test.go

```diff
@@ -895,23 +895,32 @@ func TestNodeUnpublishVolume(t *testing.T) {
 }
 
 func TestNodeGetVolumeStats(t *testing.T) {
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+	mockNodeUtils := mocks.NewMockNodeUtilsInterface(mockCtl)
+	d := newTestNodeService(mockNodeUtils, nil, nil)
+	volId := "someStorageType:vol-test"
+	volumePath := "/test/path"
+	stagingTargetPath := "/staging/test/path"
+	expErrCode := codes.NotFound
 
-	req := &csi.NodeGetVolumeStatsRequest{}
-
-	d := newTestNodeService(nil, nil, nil)
-
-	expErrCode := codes.Unimplemented
+	mockNodeUtils.EXPECT().IsPathExists("/host" + volumePath).Return(false)
+	mockNodeUtils.EXPECT().GetPodPath(volumePath).Return("/host" + volumePath)
+	req := &csi.NodeGetVolumeStatsRequest{
+		VolumeId:          volId,
+		VolumePath:        volumePath,
+		StagingTargetPath: stagingTargetPath,
+	}
 
 	_, err := d.NodeGetVolumeStats(context.TODO(), req)
-	if err == nil {
-		t.Fatalf("Expected error code %d, got nil", expErrCode)
-	}
-	srvErr, ok := status.FromError(err)
-	if !ok {
-		t.Fatalf("Could not get error status code from error: %v", srvErr)
-	}
-	if srvErr.Code() != expErrCode {
-		t.Fatalf("Expected error code %d, got %d message %s", expErrCode, srvErr.Code(), srvErr.Message())
+	if err != nil {
```

Comment:
shouldn't the test fail if `err` is `nil`?
in addition, please add a "happy flow" test case, to see the (fake) stats values actually exist in the response volume usage

Reply:
another test case I just thought of: assert unix stat failure leads to a CSI internal status code error

Reply:
added an handle to nil error

Reply:
done both of the unit tests